central/authprovider/datastore/telemetry.go

```diff
@@ -0,0 +1,45 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	groupDataStore "github.com/stackrox/rox/central/group/datastore"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/telemetry/phonehome"
+)
+
+// Gather auth provider names and number of groups per auth provider.
+func Gather(ctx context.Context) (phonehome.Properties, error) {
+	ctx = sac.WithAllAccess(ctx)
+	props := make(phonehome.Properties)
+
+	providers, err := Singleton().GetAllAuthProviders(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get AuthProviders")
+	}
+
+	providerIDNames := make(map[string]string)
+	providerNames := make([]string, 0, len(providers))
+	for _, provider := range providers {
+		providerIDNames[provider.GetId()] = provider.GetName()
+		providerNames = append(providerNames, provider.GetName())
+	}
+	props["Auth Providers"] = providerNames
+
+	groups, err := groupDataStore.Singleton().GetAll(ctx)
+	if err != nil {
+		return props, errors.Wrap(err, "failed to get Groups")
```

Comment:
I understand it violates the expectation of nil result in case of error. Here though we can still get the auth providers, which is more useful than nothing.